Firestore/core/src/firebase/firestore/local/local_serializer.cc

```diff
@@ -106,8 +114,14 @@ std::unique_ptr<model::MaybeDocument> LocalSerializer::DecodeMaybeDocument(
         // claim that if both are set on the wire, the last one wins.
         document = nullptr;
 
-        // TODO(rsgowman): Parse the no_document field.
-        abort();
+        // TODO(rsgowman): If multiple 'no_document' values are found, we should
+        // merge them (rather than using the last one.)
+        no_document =
+            reader->ReadNestedMessage<std::unique_ptr<model::NoDocument>>(
+                [&](Reader* reader) -> std::unique_ptr<model::NoDocument> {
```

Comment:
Sorry if we discussed this already, but I think the return type should be optional for this lambda. Does the compiler reject this code without explicitly specifying the return type?

Reply:
> Sorry if we discussed this already, 

I don't think we have? (Or if so, I've forgotten)

> but I think the return type should be optional for this lambda. Does the compiler reject this code without explicitly specifying the return type?

Compiler is happy without it. Generally, I like to include it (for the same reason I avoid 'auto') but in this case, the return type is literally *right there* already, so this is redundant. Removed.